crates/nu-test-support/src/fs.rs

```diff
@@ -233,18 +233,21 @@ pub fn root() -> PathBuf {
 }
 
 pub fn binaries() -> PathBuf {
-    let mut build_type = "debug".to_string();
-    if !cfg!(debug_assertions) {
-        build_type = "release".to_string()
-    }
-    if let Ok(target) = std::env::var("NUSHELL_CARGO_TARGET") {
-        build_type = target;
-    }
+    let build_target = std::env::var("CARGO_BUILD_TARGET").unwrap_or(String::new());
+
+    let profile = if let Ok(env_profile) = std::env::var("NUSHELL_CARGO_PROFILE") {
+        env_profile
+    } else if cfg!(debug_assertions) {
+        "debug".into()
+    } else {
+        "release".into()
+    };
 
     std::env::var("CARGO_TARGET_DIR")
-        .ok()
-        .map(|target_dir| PathBuf::from(target_dir).join(&build_type))
-        .unwrap_or_else(|| root().join(format!("target/{}", &build_type)))
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| root().join("target"))
+        .join(build_target)
+        .join(profile)
```

Comment:
I'm a bit unclear on how this fix is related to the platform support.

For unrelated changes, we generally do separate, targeted fixes (just to make things a bit easier to review)

Reply:
I explained this in the PR description—when the `CARGO_BUILD_TARGET` is set explicitly (which is common on Termux due to the increasing use of Rust in Python packages), it is part of the build output path, so e.g. on my device, the path to the `nu` binary is

`target/aarch64-linux-android/debug/nu`

rather than just

`target/debug/nu`

It's true that this is more general of an issue than just Android support, but I think more likely on Android due to happenstance.

I can split this off into another PR if you'd like though.